src/olympia/editors/management/commands/auto_approve.py

```diff
@@ -0,0 +1,139 @@
+
+        if min_approved_updates <= 0 or max_average_daily_users <= 0:
+            # Auto approval are shut down if one of those values is not present
+            # or <= 0.
+            url = '%s%s' % (
+                settings.SITE_URL,
+                reverse('admin:zadmin_config_changelist'))
+            raise CommandError(
+                'Auto-approvals are deactivated because either '
+                'AUTO_APPROVAL_MAX_AVERAGE_DAILY_USERS or '
+                'AUTO_APPROVAL_MIN_APPROVED_UPDATES have not been '
+                'set or were set to 0. Use the admin tools Config model to '
+                'set them by going to %s.' % url)
+
+        stats = Counter()
+        qs = self.fetch_candidates()
+        stats['total'] = len(qs)
+
+        successful_verdict = (
+            amo.WOULD_HAVE_BEEN_AUTO_APPROVED if dry_run
+            else amo.AUTO_APPROVED)
+
+        for version in qs:
+            # Is the addon already locked by a reviewer ?
+            if get_reviewing_cache(version.addon.pk):
+                stats['locked'] += 1
+                continue
+
+            # Lock the addon for ourselves, no reviewer should touch it.
+            set_reviewing_cache(version.addon.pk, settings.TASK_USER_ID)
+
+            # If admin review or more information was requested, skip this
+            # version, let a human handle it.
+            if version.addon.admin_review or version.has_info_request:
+                stats['flagged'] += 1
+                clear_reviewing_cache(version.addon.pk)
+                continue
+
+            try:
+                log.info('Processing %s version %s...',
+                         unicode(version.addon.name), unicode(version.version))
+                summary, info = AutoApprovalSummary.create_summary_for_version(
+                    version, max_average_daily_users=max_average_daily_users,
+                    min_approved_updates=min_approved_updates,
+                    dry_run=dry_run)
+                log.info('Auto Approval for %s version %s: %s',
+                         unicode(version.addon.name),
+                         unicode(version.version),
+                         summary.get_verdict_display())
+                stats.update({k: int(v) for k, v in info.items()})
+                if summary.verdict == successful_verdict:
+                    stats['auto_approved'] += 1
+                # FIXME: implement auto-approve if verdict is amo.AUTO_APPROVED
+
+            except (AutoApprovalNotEnoughFilesError,
+                    AutoApprovalNoValidationResultError):
+                log.info(
+                    'Version %s was skipped either because it had no '
+                    'file or because it had no validation attached.', version)
+                stats['error'] += 1
+                clear_reviewing_cache(version.addon.pk)
+                continue
+            clear_reviewing_cache(version.addon.pk)
```

Comment:
The regular review view does clear the lock when processing a `POST` though, so I think we should do the same here.